### PR TITLE
Bug 1403975: scheduled change modal hangs when no "when" is set and asap is not clicked

### PR DESCRIPTION
**Pull request by collin5**
Fixes `Error: b.when is undefined` which occurs when adding scheduled changes without setting time/date.

### CODE REVIEW

**Comment by collin5** [comment]
@mozbhearsum Thanks for the feedback.
I also had to check if `data.when` in `null` in rule scheduled changes backend to be consistent with using null as this is the case for release schedule changes

**Comment by collin5** [comment]
@mozbhearsum PR should be now ready for another review. Thanks

File: auslib/web/admin/views/rules.py [hunk 1]
```diff
@@ -200,6 +200,9 @@ def __init__(self):
 
     @requirelogin
     def _post(self, transaction, changed_by):
+        if connexion.request.get_json().get("when", None) is None:
+            return problem(400, "Bad Request", "'when' cannot be set to null when scheduling a new change "
+                                               "for a Release")
```
Comment (mozbhearsum) [inline auslib/web/admin/views/rules.py]: I think this should be "Rule", not "Release"?

File: auslib/test/admin/views/test_rules.py [hunk 2]
```diff
@@ -1206,10 +1208,11 @@ def testGetScheduledChangesWithCompleted(self):
         self.assertEquals(json.loads(ret.data), expected)
 
     def testAddScheduledChangeExistingRule(self):
+        future_time = int(time.mktime((datetime.datetime.now() + datetime.timedelta(minutes=5)).timetuple()) * 1000)
```
Comment (mozbhearsum) [inline auslib/test/admin/views/test_rules.py]: This works, but please following the existing pattern for this by using mock.patch instead, like: https://github.com/andela/balrog/blob/20ee523439c183b9581812f23afe2a69f5d2b706/auslib/test/admin/views/test_rules.py#L1289.

File: auslib/web/admin/views/rules.py (hunk 1 above)
Comment (collin5) [reply auslib/web/admin/views/rules.py]: Thank you, I had completely missed that one.

**Comment by collin5** [comment]
@mozbhearsum PR should be ready now, thanks again 👍

**Pull request merged**